### PR TITLE
Chainlink solo integration

### PR DESCRIPTION
**Pull request by piobab**
Chainlink Solo:

System queries the Chainlink price.
If the Chainlink price is USD denominated, the system converts the price to UST using the Chainlink UST/USD price feed. The remaining UST denominated price is Chainlink UST Price.

### CODE REVIEW

File: contracts/mars-oracle/src/contract.rs [hunk 1]
```diff
@@ -19,6 +19,8 @@ use crate::{AstroportTwapSnapshot, Config, PriceSourceChecked, PriceSourceUnchec
 use self::helpers::*;
 use astroport::pair::TWAP_PRECISION;
 
+pub const DEFAULT_DEVIATION_PERCENTAGE: u64 = 15;
```
Comment (alienHunterOnMars) [inline contracts/mars-oracle/src/contract.rs:22]: looks like we are not using this anywhere, maybe we can remove it ?

**Review by alienHunterOnMars — DISMISSED** [review]

File: contracts/mars-oracle/Cargo.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "mars-oracle"
-version = "1.0.1"
```
Comment (dancreee) [inline contracts/mars-oracle/Cargo.toml:3]: This should be bumped to 1.0.3?

File: contracts/mars-oracle/Cargo.toml [hunk 3]
```diff
@@ -38,6 +38,8 @@ thiserror = "1.0.23"
 
 astroport = "1.0"
 
+chainlink-terra = { git = "https://github.com/smartcontractkit/chainlink-terra", package = "proxy-ocr2", version = "1.0.0", default-features = false, features = ["library"] }
```
Comment (dancreee) [inline contracts/mars-oracle/Cargo.toml:41]: is there a reason you have added the package in both mars-oracle and mars-core, whereas with the other oracle sources you have added the package only to mars-core and used it in mars-oracle through mars-core?

**Review by dancreee — DISMISSED** [review]

File: contracts/mars-oracle/src/contract.rs (hunk 1 above)
Comment (piobab) [reply contracts/mars-oracle/src/contract.rs]: Yes we can, left after extracting chainlink solo from previous PR.

File: contracts/mars-oracle/Cargo.toml (hunk 2 above)
Comment (piobab) [reply contracts/mars-oracle/Cargo.toml:3]: Yes, but it could change if in the meantime we introduce new number :)

**Review by larry0x — DISMISSED** [review]

**Review by dancreee — APPROVED** [review]

**Review by dancreee — APPROVED** [review]

**Review by alienHunterOnMars — APPROVED** [review]

**Pull request closed without merging**